medicines/doc-index-updater/src/models.rs

```diff
@@ -83,22 +83,25 @@ impl Into<Document> for XMLDocument {
             author: self.author,
             products: self
                 .products
+                .product
                 .iter()
-                .map(move |active_substance| active_substance.name.clone())
+                .map(|product| product.to_string())
                 .collect::<Vec<String>>(),
             keywords: match self.keywords {
                 Some(kw) => Some(
-                    kw.iter()
-                        .map(move |keyword| keyword.name.clone())
+                    kw.keyword
+                        .iter()
+                        .map(|keyword| keyword.to_string())
                         .collect::<Vec<String>>(),
                 ),
                 None => None,
             },
             pl_number: self.pl_number,
             active_substances: self
                 .active_substances
+                .active_substance
                 .iter()
-                .map(move |active_substance| active_substance.name.clone())
+                .map(|active_substance| active_substance.to_string())
```

Comment:
This 3-line block is unnecessary. You're iterating just to return the initial thing.

Reply:
It's my birthday and I'll iterate if I want to!